src/backend/mod.rs

```diff
@@ -120,6 +124,27 @@ impl DNSBackend {
         Some(results)
     }
 
+    // Checks if a container is associated with only internal networks.
+    // Returns true iff a container is only present in internal networks.
+    pub fn ctr_is_internal(&self, requester: &IpAddr) -> bool {
+        let mut is_internal = true;
+
+        match self.ip_mappings.get(requester) {
+            Some(nets) => {
+                for net in nets {
+                    match self.network_is_internal.get(net) {
+                        Some(b) => is_internal &= b,
```

Comment:
you do not have to loop over all and need the is_internal tmp var

you can check if !b { return false } and then at the bottom of the function return true.
Also rename b to something like `internal`